File: sdk/go/qrd.go
package qrd

// FileReader is a Go-side QRD SDK placeholder.
type FileReader struct {
	Path      string
	MasterKey []byte
}

// FileWriter is a Go-side QRD SDK placeholder.
type FileWriter struct {
	Path   string
	Schema map[string]any
}

// InspectHeader is a placeholder that will later call the Rust core through CGO.
func InspectHeader(path string) (map[string]any, error) {
	_ = path
	return nil, errNotImplemented{}
}

// InspectFooter is a placeholder that will later call the Rust core through CGO.
func InspectFooter(path string) (map[string]any, error) {
	_ = path
	return nil, errNotImplemented{}
}

// WriteRow is a placeholder that will later call the Rust core through CGO.
func (w *FileWriter) WriteRow(row map[string]any) error {
	_ = row
	return errNotImplemented{}
}

// Finish is a placeholder that will later call the Rust core through CGO.
func (w *FileWriter) Finish() error {
	return errNotImplemented{}
}

type errNotImplemented struct{}

func (errNotImplemented) Error() string { return "CGO binding not added yet" }

File: sdk/java/src/main/java/dev/qrd/Qrd.java
package dev.qrd;

/**
 * QRD Java SDK scaffold.
 *
 * The final implementation will bridge to the Rust core through JNI.
 */
public final class Qrd {
    private Qrd() {
    }

    public static final class FileReader {
        private final String path;

        public FileReader(String path) {
            this.path = path;
        }

        public String path() {
            return path;
        }

        public String inspectHeader() {
            throw new UnsupportedOperationException("JNI binding not added yet");
        }

        public String inspectFooter() {
            throw new UnsupportedOperationException("JNI binding not added yet");
        }
    }

    public static final class FileWriter {
        private final String path;

        public FileWriter(String path) {
            this.path = path;
        }

        public String path() {
            return path;
        }

        public void writeRow(java.util.Map<String, Object> row) {
            throw new UnsupportedOperationException("JNI binding not added yet for writer at " + path);
        }

        public void finish() {
            throw new UnsupportedOperationException("JNI binding not added yet for writer at " + path);
        }
    }
}
